fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushdownFilterThroughWindow.java

```diff
@@ -0,0 +1,174 @@
+// under the License.
+
+package org.apache.doris.nereids.rules.rewrite.logical;
+
+import org.apache.doris.nereids.rules.Rule;
+import org.apache.doris.nereids.rules.RuleType;
+import org.apache.doris.nereids.rules.rewrite.OneRewriteRuleFactory;
+import org.apache.doris.nereids.trees.expressions.BinaryOperator;
+import org.apache.doris.nereids.trees.expressions.ExprId;
+import org.apache.doris.nereids.trees.expressions.Expression;
+import org.apache.doris.nereids.trees.expressions.LessThan;
+import org.apache.doris.nereids.trees.expressions.LessThanEqual;
+import org.apache.doris.nereids.trees.expressions.NamedExpression;
+import org.apache.doris.nereids.trees.expressions.SlotReference;
+import org.apache.doris.nereids.trees.expressions.WindowExpression;
+import org.apache.doris.nereids.trees.expressions.WindowFrame;
+import org.apache.doris.nereids.trees.expressions.functions.window.DenseRank;
+import org.apache.doris.nereids.trees.expressions.functions.window.Rank;
+import org.apache.doris.nereids.trees.expressions.functions.window.RowNumber;
+import org.apache.doris.nereids.trees.expressions.literal.IntegerLikeLiteral;
+import org.apache.doris.nereids.trees.plans.Plan;
+import org.apache.doris.nereids.trees.plans.logical.LogicalWindow;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Push down the 'filter' into the 'window'.
+ * It will convert the filter condition to the 'limit value' and embed it to the 'window'.
+ * But there are some restrictions, the details are explained below.
+ * For example:
+ * 'SELECT * FROM (
+ *     SELECT *, ROW_NUMBER() OVER (ORDER BY b) AS row_number
+ *     FROM t
+ * ) AS tt WHERE row_number <= 100;'
+ * The filter 'row_number <= 100' can be pushed down into the window operator.
+ * The following will demonstrate how the plan changes:
+ * Logical plan tree:
+ *                 any_node
+ *                   |
+ *                filter (row_number <= 100)
+ *                   |
+ *                window (PARTITION BY a ORDER BY b)
+ *                   |
+ *                 any_node
+ * transformed to:
+ *                 any_node
+ *                   |
+ *                filter (row_number <= 100)
+ *                   |
+ *                window (PARTITION BY a ORDER BY b) [Limit: 100]
+ *                   |
+ *                 any_node
+ */
+
+public class PushdownFilterThroughWindow extends OneRewriteRuleFactory {
+
+    @Override
+    public Rule build() {
+        return logicalFilter(logicalWindow()).then(filter -> {
+            LogicalWindow<Plan> window = filter.child();
+
+            // Check the filter conditions. Now, we currently only support
+            // simple conditions of the form 'column </<= constant'.
+            // TODO: Support more complex situations in filter conditions.
+            Set<Expression> conjuncts = filter.getConjuncts();
+            if (conjuncts.size() != 1) {
+                return filter;
+            }
```

Comment:
why check size?

Reply:
You can see the `TODO`. It maybe needs to normalize the filter conditions, for example if there is a condition contains `and` and `or`. We need to expand it. So I think it can be handled separate.